qutip/tests/solver/test_stochastic.py

```diff
@@ -0,0 +1,323 @@
+
+    options = {
+        "store_measurement": True,
+        "map": "parallel",
+        "method": method,
+        "keep_runs_results": True,
+    }
+
+    res = ssesolve(
+        H, psi0, times, sc_ops, e_ops=e_ops,
+        ntraj=ntraj, args={"w": 2}, options=options, heterodyne=heterodyne,
+        seeds=1,
+    )
+
+    for idx in range(len(e_ops)):
+        np.testing.assert_allclose(
+            res.average_expect[idx], res_ref.expect[idx], rtol=tol, atol=tol
+        )
+
+    assert len(res.measurement) == ntraj
+
+    if heterodyne:
+        assert all([
+            dw.shape == (len(sc_ops), 2, len(times)-1)
+            for dw in res.dW
+        ])
+        assert all([
+            w.shape == (len(sc_ops), 2, len(times))
+            for w in res.wiener_process
+        ])
+        assert all([
+            m.shape == (len(sc_ops), 2, len(times)-1)
+            for m in res.measurement
+        ])
+    else:
+        assert all([
+            dw.shape == (len(sc_ops), len(times)-1)
+            for dw in res.dW
+        ])
+        assert all([
+            w.shape == (len(sc_ops), len(times))
+            for w in res.wiener_process
+        ])
+        assert all([
+            m.shape == (len(sc_ops), len(times)-1)
+            for m in res.measurement
+        ])
+
+
+def test_reuse_seeds():
+    "Stochastic: smesolve: homodyne, time-dependent H"
+    tol = 0.05
+    N = 4
+    ntraj = 5
+
+    H, sc_ops = _make_system(N, "simple")
+    c_ops = [destroy(N)]
+    psi0 = coherent(N, 0.5)
+    a = destroy(N)
+    e_ops = [a.dag() * a, a + a.dag(), (-1j)*(a - a.dag())]
+
+    times = np.linspace(0, 0.1, 2)
+
+    options = {
+        "store_final_state": True,
+        "map": "serial",
+        "keep_runs_results": True,
+        "store_measurement": True,
+    }
+
+    res = smesolve(
+        H, psi0, times, sc_ops=sc_ops, e_ops=e_ops, c_ops=c_ops,
+        ntraj=ntraj, args={"w": 2}, options=options,
+    )
+
+    res2 = smesolve(
+        H, psi0, times, sc_ops=sc_ops, e_ops=e_ops, c_ops=c_ops,
+        ntraj=ntraj, args={"w": 2}, options=options,
+        seeds=res.seeds,
+    )
+
+    np.testing.assert_allclose(
+        res.wiener_process, res2.wiener_process, atol=1e-14
+    )
+
+    np.testing.assert_allclose(res.expect, res2.expect, atol=1e-14)
+
+    for out1, out2 in zip(res.final_state, res2.final_state):
+        assert out1 == out2
+
+
+@pytest.mark.parametrize("heterodyne", [True, False])
+def test_m_ops(heterodyne):
+    "Stochastic: smesolve: homodyne, time-dependent H"
+    N = 10
+    ntraj = 1
+
+    H = num(N)
+    sc_ops = [destroy(N), qeye(N)]
+    psi0 = basis(N, N-1)
+    m_ops = [num(N), qeye(N)]
+    if heterodyne:
+        m_ops = m_ops * 2
+
+    times = np.linspace(0, 1.0, 51)
+
+    options = {"store_measurement": True,}
+
+    solver = SMESolver(H, sc_ops, heterodyne=heterodyne, options=options)
+    solver.m_ops = m_ops
+    solver.dW_factors = [0.] * len(m_ops)
```

Comment:
I'm not sure what is happening in these two lines. Hopefully they are unneeded?

Reply:
Okay, let's keep this for now. It's a bit strange and different to how we do things elsewhere, but let's sort it out later when we have more use cases.